### PR TITLE
use -D_USE_METIS_5p1 in OpenSees v3.2.0 patch to correctly build on top of METIS v5.1.0

### PR DESCRIPTION
**Pull request by mckee107**
add -D_USE_METIS_5p1 flag for proper Metis.cpp function.

### CODE REVIEW

**Review by boegel — CHANGES_REQUESTED** [review]
@mckee107 You will also need to change the SHA256 checksum for this patch file in the `OpenSees` easyconfigs that use this patch...

**Comment by boegel** [comment]
@mckee107 Can you clarify which problem this fixes exactly? Any particular error messages worth mentioning, or is it more like a lack of functionality without `-D_USE_METIS_5p1`?

**Comment by mckee107** [comment]
> @mckee107 Can you clarify which problem this fixes exactly? Any particular error messages worth mentioning, or is it more like a lack of functionality without `-D_USE_METIS_5p1`?

There are `#ifdef` statements in OpenSees/SRC/graph/partitioner/Metis.cpp that rely on `_USE_METIS_5p1` to determine which version of METIS is being used. I guess there were changes to METIS functions introduced in 5.1.x, which I believe changed function inputs for atleast `METIS_PartGraphKway`. If this flag is not defined, an error will be thrown saying `Input Error: Incorrect ncuts`.

File: easybuild/easyconfigs/o/OpenSees/OpenSees-3.2.0-intel-2020a-Python-3.8.2-parallel.eb [hunk 1]
```diff
@@ -23,7 +23,7 @@ patches = [
 checksums = [
     '200f70fc86b7c4c2978eaef1aacd206cb40791fa797e1b9cc51da9633b53eb36',  # v3.2.0.tar.gz
     # OpenSees-3.2.0-add_Makefile_def_parallel.patch
-    'a5af4dc405916575a3eaea8cc78fb17c9e016cacd7f1048f906433ecd4e9da5a',
+    '96e541d9827cabe0cef64e42c6d8e6581837bc6dc4a47b476a2b76a9636d139e',
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenSees/OpenSees-3.2.0-intel-2020a-Python-3.8.2-parallel.eb:26]: @mckee107 It looks like `d9eca142b325b0c64d9e1db11f2eb994af86530c1262ed29eec44dc2d75b3ea1` is the correct SHA256 checksum after changing the patch file?

Note that you can use `eb OpenSees-3.2.0-intel-2020a-Python-3.8.2-parallel.eb  --inject-checksums --force` to correct this automatically.

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/o/OpenSees/OpenSees-3.2.0-intel-2020a-Python-3.8.2-parallel.eb (hunk 1 above)
Comment (mckee107) [reply easybuild/easyconfigs/o/OpenSees/OpenSees-3.2.0-intel-2020a-Python-3.8.2-parallel.eb:26]: I am getting `96e541d9827cabe0cef64e42c6d8e6581837bc6dc4a47b476a2b76a9636d139e` on my computer, but I may be getting it incorrectly. I can update the easyconfig to use `d9eca142b325b0c64d9e1db11f2eb994af86530c1262ed29eec44dc2d75b3ea1` if you would like.

Comment (boegel) [reply easybuild/easyconfigs/o/OpenSees/OpenSees-3.2.0-intel-2020a-Python-3.8.2-parallel.eb:26]: How exactly are you determining the checksum locally?

Comment (mckee107) [reply easybuild/easyconfigs/o/OpenSees/OpenSees-3.2.0-intel-2020a-Python-3.8.2-parallel.eb:26]: Using git bash. Tried both my local repo and the pr fork from easybuild/easyconfigs. If I'm doing it wrong, please let me know.

```
codim@DESKTOP-BVSQJN9 MINGW64 ~/Documents/GitHub/easybuild-easyconfigs/easybuild/easyconfigs/o/OpenSees (pr/12403)
$ shasum -a 256 OpenSees-3.2.0-add_Makefile_def_parallel.patch
96e541d9827cabe0cef64e42c6d8e6581837bc6dc4a47b476a2b76a9636d139e *OpenSees-3.2.0-add_Makefile_def_parallel.patch
```

Comment (boegel) [reply easybuild/easyconfigs/o/OpenSees/OpenSees-3.2.0-intel-2020a-Python-3.8.2-parallel.eb:26]: That doesn't agree with what I'm seeing when I have your `develop` branch checked out:

```
$ sha256sum easybuild/easyconfigs/o/OpenSees/OpenSees-3.2.0-add_Makefile_def_parallel.patch
d9eca142b325b0c64d9e1db11f2eb994af86530c1262ed29eec44dc2d75b3ea1  easybuild/easyconfigs/o/OpenSees/OpenSees-3.2.0-add_Makefile_def_parallel.patch

$ shasum -a 256 easybuild/easyconfigs/o/OpenSees/OpenSees-3.2.0-add_Makefile_def_parallel.patch
d9eca142b325b0c64d9e1db11f2eb994af86530c1262ed29eec44dc2d75b3ea1  easybuild/easyconfigs/o/OpenSees/OpenSees-3.2.0-add_Makefile_def_parallel.patch
```

What kind of environment are you computing the SHA256 checksum? I see MINGW64, so is this Cygwin? That could explain it, maybe...

Comment (mckee107) [reply easybuild/easyconfigs/o/OpenSees/OpenSees-3.2.0-intel-2020a-Python-3.8.2-parallel.eb:26]: I was using it on windows, MINGW64.  I tried again on the cluster login node, and got a value that agrees with you.
```
[mckee107@ada4 easyconfigs]$ shasum -a 256 o/OpenSees/OpenSees-3.2.0-add_Makefile_def_parallel.patch
d9eca142b325b0c64d9e1db11f2eb994af86530c1262ed29eec44dc2d75b3ea1  o/OpenSees/OpenSees-3.2.0-add_Makefile_def_parallel.patch
```

I'll update the file

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3519.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/58fd81aadec25ba11eb2f5c2a5651e7f for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2665.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/16e371fcce31bd78d7fb9146acb6dfa6 for a full test report.

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Going in, thanks @mckee107!

**Pull request merged**